cli/sarifanalysisreport.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2022 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "sarifanalysisreport.h"
+
+#include <picojson.h>
+
+SARIFAnalysisReport::SARIFAnalysisReport( std::string versionNumber)
+    : mVersionNumber(std::move(versionNumber)) {}
+
+void SARIFAnalysisReport::addFinding(const ErrorMessage &msg) {
+    std::map<std::string, std::vector<ErrorMessage>>::iterator it = mFindings.insert(mFindings.begin(), {msg.id, std::vector<ErrorMessage>()});
```

Comment:
is this code the same as :

    mFindings[msg.id].push_back(msg);

Reply:
Thanks for the suggestion. I've fixed it now :)